src/lib/y2storage/planned/device.rb

```diff
@@ -60,8 +61,39 @@ def self.to_string_attrs
         [:reuse]
       end
 
+      # Returns the (possibly processed) device to be used for the planned
+      # device.
+      #
+      # FIXME: temporary API. It should be improved.
+      def final_device!(plain_device)
+        plain_device.encrypted? ? plain_device.encryption : plain_device
```

Comment:
well, all this is quite highly tight to encryption. I worry what happen if there is something similar like encryption child. How much code have to be adapted?